var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -37,61 +42,67 @@ class ModelHippocampus(SimModel):
 
     phases = ('build', 'install')
 
-    mech_name = ''
+    mech_name = None
     """The name of the mechanism, defined in subclasses"""
 
     def build(self, spec, prefix):
-        """Build phase
-        """
-        link_flag = '-Wl,-rpath,' + prefix.lib
-        self._build_mods('mod', link_flag)
+        """Build phase"""
+        self._build_mods('mod')
+
+    @property
+    def lib_suffix(self):
+        return ('_' + self.mech_name) if self.mech_name else ''
 
     def _build_mods(self, mods_location, link_flag='', include_flag='', corenrn_mods=None):
         """Build shared lib & special from mods in a given path
         """
-        profile_flag = ' -DENABLE_TAU_PROFILER' if '+profile' in self.spec else ''
-        include_flag += profile_flag
+        # pass include and link flags for all dependency libraries
+        # Compiler wrappers are not used to have a more reproducible building
+        dep_names = set(self.spec.dependencies_dict('link').keys())
+        for dep in dep_names:
+            if self.spec[dep].prefix in ('/usr', '/usr/local'):
+                raise Exception('Dependency lib "' + dep + '" coming from system prefix')
```

Comment:
why exception? we often use dependencies like hdf5 from /usr.

Reply:
I can remove this, but I think it's pretty unusual for our central neurodamus deployments to use /usr libs, and it already caught me undesirable situations

Reply:
I guess that it is an issue when installed in our local systems. For example I had this issue with `zlib` which was pointing to the `/usr` folder and was also set as `Buildable: False` in my `.spack/packages.yaml`, so I had to change this as well. I don't know if other users have the `packages.yaml` file and this exception can create problems for them

Reply:
Fair enough. This was mostly to have the dependencies correctly set, so we can remove.